openadmet/toolkit/filtering/physchem_filters.py

```diff
@@ -0,0 +1,416 @@
+import numpy as np
+import pandas as pd
+from openadmet.toolkit.chemoinformatics.rdkit_funcs import canonical_smiles
+from medchem.catalogs import catalog_from_smarts
+
+from rdkit.Chem.MolStandardize import rdMolStandardize
+from rdkit.Chem import MolStandardize
+from rdkit.Chem import AllChem
+from rdkit import Chem
+from tqdm import tqdm
+tqdm.pandas()
+
+import datamol as dm
+
+from pydantic import BaseModel, Field
+from openadmet.toolkit.filtering.filter_base import BaseFilter
+from typing import ClassVar
+
+# - ProximityFilter and add test for it
+
+class SMARTSFilter(BaseFilter):
+    """
+    Filter class to filter a DataFrame based on SMARTS patterns.
+
+    """
+    smarts_list: list
+    names_list: list
+    include: bool = True # whether to filter for or against smarts
+
+    names_column:str
+    mark_column: ClassVar[str] = "passed_smarts_filter"
+
+    def calculate(self,
+                  df: pd.DataFrame,
+                  smiles_column: str = "OPENADMET_CANONICAL_SMILES",
+                  mol_column: str = "mol") -> pd.DataFrame:
+        """
+        Run the SMARTS filter on the DataFrame.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            The input DataFrame to be filtered.
+        smarts_list : list
+            The list of SMARTS patterns to match in the DataFrame.
+        names_list : list
+            The list of names corresponding to the SMARTS patterns.
+        smarts_col : str
+            The column name to write to containing .
+        mol_col : str
+            The column name containing the RDKit Mol objects (default is 'mol').
+
+        Returns
+        -------
+        pandas.DataFrame
+            The DataFrame with .
+        """
+
+        df = self.set_mol_column(df=df, smiles_column=smiles_column, mol_column=mol_column)
+
+        df[self.names_column] = df[mol_column].apply(
+            lambda x: self.match_smarts_by_catalog(x, self.smarts_list, self.names_list)
+        )
+
+        return df
+
+    def filter(self,
+               df: pd.DataFrame,
+               smiles_column: str = "OPENADMET_CANONICAL_SMILES",
+               mol_column: str = "mol",
+               mode: str = "mark",
+               calculate: bool = "True") -> pd.DataFrame:
+        """
+        Run the SMARTS filter on the DataFrame.
+
+        Parameters
+        ----------
+        df : pandas.DataFrame
+            The input DataFrame to be filtered.
+        mode : str
+            Either "mark" or "remove". If "mark", the filter will mark the rows that meet the criteria
+            either True or False. If "remove", the filter will remove the rows that meet the criteria.
+
+        Returns
+        -------
+        pandas.DataFrame
+            The filtered DataFrame.
+        """
+
+        if calculate:
```

Comment:
Unless I am missing something, I would remove the calculate logic, just auto-calculate every time.